plugins/source/azure/codegen/recipes/web.go

```diff
@@ -23,11 +23,12 @@ func Web() []Resource {
 	var vnetInfoResource = resourceDefinition{
 		azureStruct:  &web.VnetInfo{},
 		listFunction: "GetVnetConnection",
-		listFunctionArgsInit: []string{"site := parent.Item.(web.Site)", `if site.SiteConfig == nil {
-			               return nil
-			       }
-				   `},
-		listFunctionArgs: []string{"*site.ResourceGroup", "*site.Name", "*site.SiteConfig.VnetName"},
+		listFunctionArgsInit: []string{"site := parent.Item.(web.Site)", `vnetName := funk.Get(site, "SiteConfig.VnetName")
```

Comment:
why not just do `if site.SiteConfig == nil || site.SiteConfig.VnetName == nil { return nil }` ?

Reply:
@erezrokah yes but it also removes type safety a bit (you have to do `vnetname.(*string)` which might get broken in an azure "upgrade") keeping it basic/regular also adds readability in a sense that a maintainer doesn't have to know about funk.Get.

Reply:
Good point, I'm counting on the mock tests to validate any changes to the Azure API.
Anyway I don't have a strong opinion on this, so I'll go ahead and update the code to your suggestion